### PR TITLE
Lists - enhancements

### PR DESCRIPTION
**Pull request by tethryus**
add.sl, renamed to add element.sl and the operation, now supports sets too.

### CODE REVIEW

**Comment by dashapaster** [comment]
@tethryus , can you add to name _set.sl(for example add_element_set.sl ) or maybe put it in new folder for sets, because in folder lists i created operation add.sl ( for lists:) and it will be confusing

File: content/io/cloudslang/base/lists/add_element.sl [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  action:
+    python_script: |
+      error_message = ""
+
+      if unique_element in list and unique_element != '':
+        error_message = ("Already in list")
+      elif unique_element not in list and unique_element != '':
+        list.append(unique_element)
+      else:
+        unique_element == '' and element != ''
```
Comment (mihaitusa) [inline content/io/cloudslang/base/lists/add_element.sl]: unique_element == '' and element != '' is actually a condition?

File: remove_object_from_json_array.sl [hunk 2]
```diff
@@ -0,0 +1,57 @@
+#   (c) Copyright 2015 Hewlett-Packard Development Company, L.P.
```
Comment (mihaitusa) [inline remove_object_from_json_array.sl]: Why JSONs related functionalities are in PR related to lists?

File: test_remove_object_from_json_array.sl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            - index
+        publish:
+          - json_output
+        navigate:
+          SUCCESS: test_equality
+          FAILURE: CREATEFAILURE
+    - test_equality:
+        do:
+          equals:
+            - json_input1: ${ json_output }
```
Comment (mihaitusa) [inline test_remove_object_from_json_array.sl]: How can you tell in here if the 2 JSON objects are equal? If the elements within the JSONs are not in the same order the condition that you put it in here is still valid?

**Comment by mihaitusa** [comment]
@tethryus  Please handle the comments!

File: content/io/cloudslang/base/lists/add_element.sl (hunk 1 above)
Comment (tethryus) [inline content/io/cloudslang/base/lists/add_element.sl]: Yes, it is a condition. I will change the code to reflect it.

File: remove_object_from_json_array.sl (hunk 2 above)
Comment (tethryus) [inline remove_object_from_json_array.sl]: Will remove the json part from the commit.

File: test_remove_object_from_json_array.sl (hunk 3 above)
Comment (tethryus) [inline test_remove_object_from_json_array.sl]: I will change the validation logic. Thanks for the input so far and raising the right issues in order to make the code more effective.

**Comment by mihaitusa** [comment]
@tethryus  If all tests are passing then IMHO we are ready for merge...

**Pull request merged**